### DISCUSSION TITLE
How do I emit a message to specific room to my socket.io client in js ?

### DISCUSSION THREAD

**Discussion by Axnjr** [Totodore/socketioxide#272] [Q&A] [answered]
I am using socket.io client in javascript for sending messages to my socketiooxide server. 
 > https://github.com/dreamsofcode-io/chatly-web

I have a few rooms, how do I emit messages from the server to a specific room?

I tried `io.to().emit()` but nothing happend

**Comment by Totodore** [comment, marked as answer]
Did you make your socket join the given rooms ? 
https://docs.rs/socketioxide/latest/socketioxide/socket/struct.Socket.html#method.join

Once it is done you can use `to("my_room")`  to emit to a given room.

**Reply by Axnjr** [reply]
just had another question how efficient is it to use socketiooxide rather than socket.io in nodejs ?
